core/unit_test/TestMathematicalFunctions.hpp

```diff
@@ -221,40 +221,64 @@ using math_binary_function_return_type_t = typename math_binary_function_return_
 // clang-format on
 
 struct FloatingPointComparison {
-  double rel_tol_ = 1e-6;
-  double abs_tol_ = 1e-12;
+ private:
+  template <class T>
+  KOKKOS_FUNCTION double rel_tol(T) const {
+    return DBL_EPSILON;
+  }
+  KOKKOS_FUNCTION
+  double rel_tol(float) const { return FLT_EPSILON; }
+  template <class T>
+  KOKKOS_FUNCTION double abs_tol(T) const {
+    return DBL_EPSILON;
+  }
```

Comment:
Any reason why you did not provide a `float` overload for `abs_tol`?

I am asking because I was looking into having just one `eps()` function.